### PR TITLE
feat(OrganizationsSdkAdaper): add organizations adapter

### PR DESCRIPTION
**Pull request by cricku**
Implement organizations adapter

https://jira-eng-gpk2.cisco.com/jira/browse/SPARK-220451

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -144,4 +144,4 @@
       ]
     ]
   }
-}
+}
```
Comment (cipak) [inline package.json:147]: do we have a convention on end-of-file newlines?

File: src/OrganizationsSDKAdapter.test.js [hunk 2]
```diff
@@ -0,0 +1,59 @@
+import {isObservable} from 'rxjs';
+
+import OrganizationsSDKAdaper from './OrganizationsSDKAdapter';
```
Comment (cipak) [inline src/OrganizationsSDKAdapter.test.js:3]: adapter:) also on line 8

File: package.json [hunk 3]
```diff
@@ -33,7 +33,7 @@
   },
   "dependencies": {
     "@webex/common": "^1.105.0",
-    "@webex/component-adapter-interfaces": "^1.11.0"
+    "@webex/component-adapter-interfaces": "^1.13.0"
```
Comment (lalli-flores) [inline package.json:36]: Things move fast around here! I think latest is now `1.16.0`. Let's use latest so we don't have to update again later. Also, please use a separate `fix` commit for any changes related to dependencies

File: src/OrganizationsSDKAdapter.js [hunk 4]
```diff
@@ -0,0 +1,70 @@
+import {ReplaySubject, defer} from 'rxjs';
+import {map} from 'rxjs/operators';
+import {deconstructHydraId} from '@webex/common';
+import {OrganizationsAdapter} from '@webex/component-adapter-interfaces';
+
+// TODO: Figure out how to import JS Doc definitions and remove duplication.
+/**
+ * A Member object that is part of a membership
```
Comment (lalli-flores) [inline src/OrganizationsSDKAdapter.js:8]:
```suggestion
 * A set of people in Webex.
```

File: src/OrganizationsSDKAdapter.js [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+ *
+ * @external Organization
+ * @see {@link https://github.com/webex/component-adapter-interfaces/blob/master/src/OrganizationsAdapter.js#L6}
+ */
+
+/**
+ * The `OrganizationsSDKAdapter` is an implementation of the `OrganizationsAdapter` interface.
+ * This adapter utilizes the Webex JS SDK to fetch data about a organization.
```
Comment (lalli-flores) [inline src/OrganizationsSDKAdapter.js:16]:
```suggestion
 * This adapter utilizes the Webex JS SDK to fetch data about an organization.
```

File: src/OrganizationsSDKAdapter.js [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+    super(datasource);
+
+    this.organizationObservables = {};
+  }
+
+  /**
+   * Fetches the organization data and returns in the shape required by adapter.
+   *
+   * @param {string} orgID ID of the organization for which to fetch data
+   * @returns {Promise<Organization>} Information about the organization of the given ID
```
Comment (lalli-flores) [inline src/OrganizationsSDKAdapter.js:31]:
```suggestion
   * @returns {Promise.<Organization>} Information about the organization of the given ID
```

File: src/OrganizationsSDKAdapter.js [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+ */
+export default class OrganizationsSDKAdapter extends OrganizationsAdapter {
+  constructor(datasource) {
+    super(datasource);
+
+    this.organizationObservables = {};
+  }
+
+  /**
+   * Fetches the organization data and returns in the shape required by adapter.
```
Comment (lalli-flores) [inline src/OrganizationsSDKAdapter.js:28]: This method doesn't seem to be doing any shaping 😉 
```suggestion
   * Returns a promise to organization data from Webex.
```

File: src/OrganizationsSDKAdapter.js [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+   * @returns {external:Observable.<Organization>} Observable stream that emits organization data
+   */
+  getOrg(ID) {
+    if (!(ID in this.organizationObservables)) {
+      // use ReplaySubject cause we don't need to set an initial value
+      this.organizationObservables[ID] = new ReplaySubject(1);
+
+      defer(() => this.fetchOrganization(ID)).pipe(
+        map((response) => ({
+          ID: deconstructHydraId(response.id).id,
```
Comment (lalli-flores) [inline src/OrganizationsSDKAdapter.js:55]: Why are you deconstructing the ID? You should send back the "Hydra ID"

File: src/OrganizationsSDKAdapter.js [hunk 9]
```diff
@@ -0,0 +1,70 @@
...
+          name: response.displayName,
+        })),
+      ).subscribe(
+        (organization) => {
+          this.organizationObservables[ID].next(organization);
+        },
+        () => {
+          this.organizationObservables[ID].error(new Error(`Could't find organization with ID ${ID}`));
```
Comment (lalli-flores) [inline src/OrganizationsSDKAdapter.js:63]:
```suggestion
          this.organizationObservables[ID].error(new Error(`Could't find organization with ID "${ID}"`));
```

File: src/OrganizationsSDKAdapter.test.js [hunk 10]
```diff
@@ -0,0 +1,59 @@
...
+            ID: 'organizationID',
+            name: 'Cisco Systems, Inc.',
+          });
+          done();
+        },
+      );
+    });
+  });
+
+  describe('getorg() errors', () => {
```
Comment (lalli-flores) [inline src/OrganizationsSDKAdapter.test.js:46]: No need to add this extra describe, unless there was some special setup, but that doesn't seem to be the case

File: package.json (hunk 3 above)
Comment (cricku) [reply package.json]: @lalli-flores do you mean just updating the version to `1.16.0` or using the `latest` tag  e.g: `"@webex/component-adapter-interfaces": "latest"` ?

File: src/OrganizationsSDKAdapter.js (hunk 8 above)
Comment (lalli-flores) [reply src/OrganizationsSDKAdapter.js]: Right, the "hydra IDs" are the public API IDs, which are an encoded version of the backend UUIDs + metadata. When interacting with some internal services we need to deconstruct it but not when exposing it outwards.

**Review by lalli-flores — COMMENTED** [review]
Code looks good! But seems like using `latest` as the version does not work. Maybe try setting it to `^1.16.0`?

File: package.json [hunk 11]
```diff
@@ -33,7 +33,7 @@
   },
   "dependencies": {
     "@webex/common": "^1.105.0",
-    "@webex/component-adapter-interfaces": "^1.11.0"
+    "@webex/component-adapter-interfaces": "latest"
```
[the remark this answers is not in the document]
Comment (cricku) [reply package.json:36]: locally it's working fine but when the checks are run in Circle CI seems that using the tag `latest` on a package doesn't install that package and that's why the linter fails

**Review by lalli-flores — APPROVED** [review]

**Pull request merged**